Feat/add circleci pipeline

figuerascarlos opened the pull request:
This PR close #6

Automatics releases can be tackled in another PR.  

WDYT @fllaca  @hrodes

figuerascarlos committed:
remove space in example name folder

figuerascarlos committed:
circleci pipeline

figuerascarlos committed:
build only tagged versions

hrodes commented on .circleci/config.yml:
```diff
@@ -0,0 +1,54 @@
+      - run:
+          name: Install golangci-lint
+          command: |
+            curl -sLo golangci-lint.tar.gz https://github.com/golangci/golangci-lint/releases/download/v${GOLANGCI_LINT_VERSION}/golangci-lint-${GOLANGCI_LINT_VERSION}-linux-amd64.tar.gz
```
If we are going to download in every build may be it is better to just download and place it in the repo... Thoughts on this?

fllaca replied:
I would create our own version from `circleci/golang:1.14` with our custom tooling. This image should be public as well.

hrodes commented on Makefile:
```diff
@@ -32,7 +32,7 @@ define compile
 .PHONY: build
-build: _bin lint test
+build: _bin lint
 	$(call compile, darwin)
```
should we add 32 bit support?

fllaca replied:
I think we can skip it for now.

figuerascarlos replied:
whats that of 32 bits ?? hahahah. 
 
for this milestone I don't think so, may we can add the issue and let that for future versions. WDYT ?

fllaca replied:
agree, for first version I think 64 bits should be enough

fllaca commented on .circleci/config.yml:
```diff
@@ -0,0 +1,54 @@
+            echo 'export PATH="~/bin:$PATH"'
+      - run:
+          name: Go verify
+          command: go mod verify
```
should `go mod verify` part of `make test`?

fllaca reviewed (commented):
lgtm, just minor thingies

figuerascarlos committed:
move go mod ops to makefile

some useful go mods operations are moved inside makefile to ensure all
dependencies are correct
